lychee-lib/src/client.rs

```diff
@@ -262,35 +225,32 @@ impl Client {
         }
     }
 
-    fn extract_github(&self, url: &str) -> Result<(String, String)> {
-        let re = Regex::new(r#"^(https?://)?(www.)?github.com/(?P<owner>[^/]*)/(?P<repo>[^/]*)"#)?;
-        let caps = re.captures(&url).context("Invalid capture")?;
-        let owner = caps.name("owner").context("Cannot capture owner")?;
-        let repo = caps.name("repo").context("Cannot capture repo")?;
-        Ok((owner.as_str().into(), repo.as_str().into()))
-    }
-
-    pub async fn check_mail(&self, address: &str) -> bool {
-        let input = CheckEmailInput::new(vec![address.to_string()]);
+    pub async fn check_mail(&self, uri: &Uri) -> Status {
+        let input = CheckEmailInput::new(vec![uri.as_str().to_owned()]);
         let results = check_email(&input).await;
-        let result = results.get(0);
-        match result {
-            None => false,
-            Some(result) => {
-                // Accept everything that is not invalid
-                !matches!(
-                    result.is_reachable,
-                    check_if_email_exists::Reachable::Invalid
-                )
-            }
+        let result = results.get(0).unwrap();
+
+        // TODO: We should not be using a HTTP status code for mail
```

Comment:
Guess this TODO can be removed now, because you fixed it.

Reply:
Will remove in next commit